Add windows bosh agent support

nmuesch opened the pull request:
*Note: Please remember to review the Datadog [Contribution Guidelines](https://github.com/DataDog/dd-agent/blob/master/CONTRIBUTING.md)
if you have not yet done so.*

### What does this PR do?

Adds a new job/package called `dd-agent-windows` that will work on windows stemcells using the Windows Agent Zip Build

Also updated the build script to support multiple jobs (builds dd-agent job by default, but allows specification of individual job)

This windows version also uses windows services to control the agent, instead of running the binaries directly in the foreground. This requires some manual additions to the registry + creation of services. 

### Motivation

Supporting Windows Stemcells

### Additional Notes

Manually deployed this to a windows stemcell and seeing the agent services start as expected. 
The documentation around bosh releases states that different OSes should likely be different bosh releases - https://bosh.io/docs/jobs/ Due to this and the monit files handling services differently, the windows support is a new package.

zippolyte commented on jobs/dd-agent-windows/templates/config/datadog.yaml.erb:
```diff
@@ -0,0 +1,534 @@
+#   Only change if the defaults are causing issues.
+#   max_per_message:
+#   Overrides the path to the Agent bin used for getting the hostname. The default is usually fine.
+  dd_agent_bin: "/var/vcap/jobs/dd-agent-windows/packages/dd-agent-windows/Datadog Agent/bin/agent.exe"
```
no `\` here ?

zippolyte commented on jobs/dd-agent-windows/templates/bin/drain.ps1:
```diff
@@ -0,0 +1 @@
+Start-Process -NoNewWindow -Wait "\var\vcap\packages\dd-agent-windows\Datadog\bin\agent.exe" -ArgumentList "stopservice"
```
Should we do this part about printing an int to the output ? https://bosh.io/docs/drain/#script-implementation

nmuesch commented:
I've reworked this PR a bit based on a different Windows Agent package, and updated the PR description. There's some more work here, like the build scripts, but the packaging is mostly working so far.
